docs/page-objects/index.rst

```diff
@@ -0,0 +1,121 @@
+.. _page-objects:
+
+============
+Page objects
+============
+
+A page object is a code wrapper for a webpage that implements the logic to
+parse the raw webpage data into structured data.
+
+To use web-poet, :ref:`define page object classes <page-object-classes>` for
+your target websites, and :ref:`get the output item <output-item>` using a
+:ref:`web-poet framework <frameworks>`.
+
+.. _page-object-classes:
+
+Defining a page object class
+============================
+
+A page object class is a Python class that:
+
+-   Subclasses :class:`~web_poet.pages.Injectable` or is registered or decorated
+    with :class:`Injectable.register <web_poet.pages.Injectable.register>`.
+
+-   Declares :ref:`typed input parameters <inputs>` in its ``__init__`` method.
+
+-   Implements a ``to_item`` method, which can be synchronous or asynchronous,
+    and returns the webpage content as an :ref:`item <items>`.
+
+For example:
+
+.. literalinclude:: page.py
+
+
+Minimizing boilerplate
+----------------------
+
+There are a few ways for you to minimize boilerplate when defining a page
+object class.
+
+For example, you can use attrs_ to remove the need for a custom ``__init__``
+method:
+
+.. _attrs: https://www.attrs.org/en/stable/index.html
+
+.. literalinclude:: attrs.py
+
+Also, it is often best to subclass :class:`~web_poet.pages.ItemPage`, which
+subclasses :class:`~web_poet.pages.Injectable` and provides an implementation of the
+``to_item`` method based on :ref:`declared fields <fields>`:
+
+.. literalinclude:: itempage.py
+
+If your page object class needs :class:`~web_poet.page_inputs.HttpResponse` as
+input, there is also :class:`~web_poet.pages.WebPage`, an
+:class:`~web_poet.pages.ItemPage` subclass that declares an
+:class:`~web_poet.page_inputs.HttpResponse` input and provides helper methods
+to use it:
+
+.. literalinclude:: webpage.py
+
+
+.. _output-item:
+
+Getting the output item
+=======================
+
+You should :ref:`include your page object classes into a page object
+registry <rules-intro>`, e.g. decorate them with :func:`~.handle_urls`:
+
+.. literalinclude:: register.py
```

Comment:
What do you think about moving .py files to a sub-folder, so that they're not mixed with .rst files?

Reply:
https://github.com/scrapinghub/web-poet/pull/152/commits/9f58ae7a428d519fae252f29c9b76ed865043988